### PR TITLE
cataclysm: Add option to save user generated data in home dir

### PR DESCRIPTION
**Pull request by sharpner**
Currently reinstalling the HEAD version will prune
templates, configs and save files. Allowing
users to store those files in $HOME will prevent this.

### CODE REVIEW

**Comment by apjanke** [comment]
Perhaps this should be the default, or only option? User data is generally better kept in the user home directory, especially under Homebrew's sandboxed model. Seems to me like this should be the default, unless there are cataclysm-specific compatibility issues, in which case we should probably introduce a `pkgshare`-based data directory.

**Comment by sharpner** [comment]
I also think it should've been the default. But changing it silently could confuse users so I didn't do it that way... what's the policy in brew about BC breaks?

**Comment by apjanke** [comment]
It's really a judgment call. Depends on how many existing users we think there are of the Homebrew package, and the extent of the breakage. Hard to say here.

@mistydemeo, you're the author on many of the previous `cataclysm` commits. What do you think?

**Comment by sharpner** [comment]
Since most people will currently lose their saves on an update anyways, I'll update the PR to use $HOME as default.

**Comment by apjanke** [comment]
That makes sense.

**Comment by sharpner** [comment]
updated :)

**Comment by zanderzhng** [comment]
this should be merged.

**Comment by sharpner** [comment]
Oh I just saw that the tests fail... :/ will need another update *sigh*

**Comment by zanderzhng** [comment]
It works with this commit, I don't know why it failed.

**Comment by tomyun** [comment]
@BrewTestBot test this please

File: cataclysm.rb [hunk 1]
```diff
@@ -42,6 +44,7 @@ def install
 
     args << "TILES=1" if build.with? "tiles"
     args << "CLANG=1" if ENV.compiler == :clang
+    args << "USE_HOME_DIR=1" if ! build.without? "home-directory"
```
Comment (tomyun) [inline cataclysm.rb]: According to the audit, `args << "USE_HOME_DIR=1" if build.with? "home-directory"` should be enough.

**Comment by tomyun** [comment]
```
Error: 1 problem in 1 formula
homebrew/games/cataclysm:
  * Don't negate 'build.with?': use 'build.without?'
```

File: cataclysm.rb (hunk 1 above)
Comment (sharpner) [inline cataclysm.rb]: yeah that's what I had at first, but that would make the "home-directory" setting optional, and not the default. My problem is that I deleted the fork xD

Comment (zanderzhng) [inline cataclysm.rb]: `args << "USE_HOME_DIR=1" if build.without? "without-home-directory"`

Comment (apjanke) [inline cataclysm.rb]: tomyun's right, you want `args << "USE_HOME_DIR=1" if build.with? "home-directory"`. That form does not make it the default; the default depends entirely on whether it's written as `option "with-home-directory"` or `option "without-home-directory"` up above.

**Comment by mistydemeo** [comment]
Ping

**Comment by mistydemeo** [comment]
Closing due to lack of response, but please feel free to reopen!

**Pull request closed without merging**